### DISCUSSION TITLE
Is the `hashFiles` function _actually_ available in the `if` context?

### DISCUSSION THREAD

**Discussion by orangejulius** [community/community#25790] [Actions]
<p>I manage an organization with many repositories. Some of them have a <code>Dockerfile</code>, and other’s don’t. We use our own Github Actions job to build Docker images for repositories that have a <code>Dockerfile</code>.</p>
<p>We’d like to be able to use identical  GitHub Actions workflow file across all repositories, to keep things easier to manage. However we only want to run the job to build Docker images in repositories that have a <code>Dockerfile</code>.</p>
<p>I looked at using the <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles" rel="noopener nofollow ugc"><code>hashFiles</code></a> function in the job’s <code>if</code> conditional, so that the job won’t run if there isn’t a Dockerfile.</p>
<p>Our workflow file ends up looking something  like this:</p>
<pre><code class="lang-auto">  build-docker-images:
    if: ${{ hashFiles('Dockerfile') }}
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v2
      - name: Build Docker images
        run: #our-code-to-build-images
</code></pre>
<p>However, this results in a startup error: <code>Unrecognized function: 'hashFiles'.</code></p>
<p>The <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability" rel="noopener nofollow ugc">context documentation</a> says that <code>hashFiles</code> <em>should</em> be available in the <code>if</code> conditional of a job, but is it actually?</p>

**Comment by pymak3r** [comment]
<p><a class="mention" href="/orangejulius">@orangejulius</a>  No luck here with hashFiles either, ever get this to work? I have the exact same use case.</p>

**Comment by orangejulius** [comment, marked as answer]
<p><a class="mention" href="/randori-jay">@randori-jay</a> funny you should just mention this a few days ago.</p>
<p>I finally got things working and trick is to add the <code>always()</code> function call to the <code>if:</code> check as well.</p>
<p>So the example above would be</p>
<pre><code class="lang-auto">build-docker-images:
    if: ${{ always() &amp;&amp; hashFiles('Dockerfile') }}
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v2
      - name: Build Docker images
        run: #our-code-to-build-images
</code></pre>
<p>You can see it in action here: <a href="https://github.com/pelias/docker-elasticsearch/blob/599cd66629452e99b3109d99ab28945dad822e4e/.github/workflows/push.yml" class="inline-onebox" rel="noopener nofollow ugc">docker-elasticsearch/push.yml at 599cd66629452e99b3109d99ab28945dad822e4e · pelias/docker-elasticsearch · GitHub</a></p>

**Reply by evmckinney9** [reply]
any explanation for why this works?

**Comment by evmckinney9** [comment]
Perhaps the [docs](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability) have been updated, but I see `hashFiles` only available in the context of `jobs.<job_id>.steps.if` and not in the context of `jobs.<job_id>.if`.

**Reply by APCBoston** [reply]
This is helpful, but also inconsistent with the "expressions" documentation at https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles, which gives absolutely zero indication that the function is anything other than universally available.
